src/share/poudriere/jail.sh

```diff
@@ -208,6 +208,30 @@ hook_stop_jail() {
 	fi
 }
 
+update_pkgbase() {
+	local make_jobs
+	local destdir="${JAILMNT}"
+
+	if [ ${JAIL_OSVERSION} -gt 1100086 ]; then
+		make_jobs="${MAKE_JOBS}"
+	fi
+
+	msg "Starting make update-packages"
+	${MAKE_CMD} -C "${SRC_BASE}" ${make_jobs} update-packages \
+		    DESTDIR=${destdir} REPODIR=${POUDRIERE_DATA}/images/${JAILNAME}-repo ${MAKEWORLDARGS}
+	if [ $? -eq 0 ]; then
+	    run_hook jail pkgbase ${POUDRIERE_DATA}/images/${JAILNAME}-repo
+	   return
+	elif [ $? -eq 2 ]; then
```

Comment:
This `$?` will always be 0 or 1 because it's the result of the `if [ $? -eq 0 ]` line.
```
# cat ret.sh
( exit 2 )
if [ $? -eq 0 ]; then
        echo 0
elif [ $? -eq 2 ]; then
        echo 2
else
        echo "lost ?"
fi
```
```
# sh ret.sh
lost ?
```
Some ways to avoid it:
```
ret=$?
if [ $ret ...
```
```
case $? in
0) # Success
 ...
 ;;
2) # Missing target, try fallback
  ...
  ;;
*) # Some other failure
  err 1 "Failed to 'make update-packages'"
  ;;
esac
```